docs/language/functions.rst

```diff
@@ -279,6 +279,38 @@ constructor.
     for storage rent and there is a minimum deposit needed for the contract. As a result,
     constructors always receive value on Parity Substrate.
 
+Overriding function selector
+____________________________
+
+When a function is called, the function selector and the arguments to the function
+are serialized (also known as abi encoded) and passed to the program. The selector
+is what the runtime program uses to determine what function was called. Usually
+the function selector is generated from a deterministic hash value of the function
+name and its arguments.
+
+The selector value can be overriden with the ``selector=hex"deadbea1"`` syntax,
+for example:
+
+.. code-block:: solidity
+
+    contract foo {
+        function get_foo() selector=hex"01020304" public return (int) {
+            return 102;
+        }
+    }
+
+Only ``public`` and ``external`` functions have a selector, and can have their
+selector overriden. On Substrate, constructors have selectors too, so they
+can also have their selector overriden. If a function overrides another one in a
+base contract, then the selector of both must match.
+
+.. warning::
+    On ewasm and Solana, changing the selector may result in a mismatch between
+    the contract metadata and the actual contract code, because the former does not explicitly store the selector, but derives it
```

Comment:
Is the issue here when you derive the metadata out of a contract already deployed, resulting in wrong metadata? If that is possible and common for Solana, then overwriting sounds super scary to me...

Reply:
This currently very messy on Solana. Currently there are many contracts without metadata which have hand-coded selectors (the idea of metadata is fairly new for Solana!). This is why we need the selector override in the first place. They need hand-written solidity `interface { ... }` definitions with the selector override.

We're adding selectors to the solana metadata to solve all these problems, and this scary warning will go away for Solana too.

Reply:
Ah I see :grin: I mean selector overwrites could maybe used for advanced things like hooks. And it is possible in `ink!` anyways. So I think in general it is still a useful feature to have.